augur/measurements.py

```diff
@@ -0,0 +1,364 @@
+"""
+Create JSON files suitable for visualization within the measurements panel of Auspice.
+"""
+import argparse
+import os
+import pandas as pd
+import sys
+
+from .utils import write_json
+from .validate import (
+    measurements as validate_measurements_json,
+    measurements_collection_config as validate_collection_config_json,
+    ValidateError
+)
+
+class HideAsFalseAction(argparse.Action):
+    """
+    Custom argparse Action that stores False for arguments passed as `--hide*`
+    and stores True for all other argument patterns.
+    """
+    def __call__(self, parser, namespace, values, option_string=None):
+        setattr(namespace, self.dest, option_string[2:6] != 'hide')
+
+
+def column_exists(collection, column, column_purpose):
+    """
+    Checks the provided column exists in the provided collection.
+    Prints an error message to stderr if the column does not exist.
+
+    Parameters
+    ----------
+    collection: pandas.DataFrame
+        Collection of measurements and metadata
+    column: str
+        Column to check exists in the collection
+    column_purpose: str
+        Purpose of provided column for detailed error message
+
+    Returns
+    -------
+    bool
+        True if column exists in collection
+    """
+    column_in_df = column in collection.columns
+    if not column_in_df:
+        print(
+            f"ERROR: Provided {column_purpose} column «{column}» does not exist in collection TSV.",
```

Comment:
A nice technique for this is to rely on Python's built-in `repr()` formatting, which you can succinctly use within f-strings by specifying a `!r` conversion like so:

```python
f"ERROR: Provided {column_purpose} column {column!r} …`
```

This improves on:

```python
f"ERROR: Provided {column_purpose} column '{column}' …`
```

since `repr()` unambiguously represents single quotes and other potentially confusing chars in the value whereas the latter doesn't.

See the [f-string spec](https://docs.python.org/3/reference/lexical_analysis.html#f-strings) for details on the conversion flags.

Reply:
(The use of guillemets or "French quotes" («») for surrounding literals in error messages probably comes from my habit of using them, which arose from wanting unambiguous (well, _less_ ambiguous) quotes in programming langs that don't have something like `repr()` close at hand. It's a hack that relies on guillemets not appearing in the value and reintroducing ambiguity, which obviously is very context dependent! `repr()` is better here. This is all to say, sorry if I've lent you this habit, Jover!)

Reply:
Good call about using `!r` as standard practice, @tsibley!

Also, I had to look up guillemets and its pronunciation, but I get where you're coming from now! :) I've learned a lot from this PR already...

Reply:
Sorry, definitely a habit carried over from SFS days 😅 
I've seen the use of `!r` in other PRs, time to pick up a new habit.

Reply:
Addressed in https://github.com/nextstrain/augur/pull/879/commits/e4138a31cba8c9885115f2c1905fde397651830f